huffman/encoder.cpp

```diff
@@ -155,32 +163,39 @@ int writeData2Code( FILE* rfp, FILE* wfp, string HT_table[] )
 
     return sum;
 }
-void writeDataSize( FILE* wfp, const int encodedSize )
-{
-    fseek( wfp, 0, SEEK_SET );
+int64_t writeDataSize( FILE* wfp, int freqArr[], string HT_table[] )
+{   
+    // Calculate encoded size
+    int64_t encodedSize = 0;
+    for ( int symIdx = 0; symIdx < NUMOFSYMBOL; symIdx++ )
+        encodedSize += (int64_t)freqArr[symIdx] * HT_table[symIdx].length();
     
     // write size
     char encodedSize_string[32] = { 0 };
-    sprintf( encodedSize_string, "%d", encodedSize );
+    sprintf( encodedSize_string, "%ld", encodedSize );
     fwrite( encodedSize_string, sizeof(char), strlen(encodedSize_string), wfp );
 
-    // write END as partition
-    char HEADER[4] = "END";
-    fwrite( HEADER, sizeof(char), strlen(HEADER), wfp );
+    return encodedSize;
 }
 void writeHuffmanTree( FILE* wfp, Node* currNode )
 {
+    // fseek( wfp, 0, SEEK_SET );
+
     // Base
     if ( currNode->endFlag == true ) {
-        char buffer[32] = {0};
-        sprintf( buffer, "%c[%d]", currNode->symbol, currNode->freq );
-        fwrite( buffer, sizeof(char)*strlen(buffer), 1, wfp );
+        fwrite( &(currNode->symbol), sizeof(char), 1, wfp );
         return;
     }
     // Recursive write
     fwrite( "( ", sizeof(char)*strlen("( "), 1, wfp );
     writeHuffmanTree( wfp, currNode->lChild);
     fwrite( ", ", sizeof(char)*strlen(", "), 1, wfp );
     writeHuffmanTree( wfp, currNode->rChild);
-    fwrite( " )", sizeof(char)*strlen(" )"), 1, wfp );
+    fwrite( ") ", sizeof(char)*strlen(") "), 1, wfp );
```

Comment:
To distinguish **delimiter** ")" and **symbol** ")".
* delimiter: 
  1 **)**
  2 **,**
  3 **(**